runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -433,6 +435,8 @@ void __quantum__qis__Rot(double phi, double theta, double omega, QUBIT *qubit, b
 
 void __quantum__qis__CNOT(QUBIT *control, QUBIT *target, bool adjoint)
 {
+    RT_FAIL_IF(control == target,
```

Comment:
Thanks! Would you be able to add a unit test for this check in `Test_LightningCoreQIS.cpp` for the coverage report?

Reply:
https://github.com/PennyLaneAI/catalyst/pull/447/commits/f156f31b67feb2db2090b4c3b8d8e3a0cb329e8e